pwndbg/heap/ptmalloc.py

```diff
@@ -144,19 +160,37 @@ def get_arena(self, arena_addr=None):
         return pwndbg.memory.poi(self.malloc_state, arena_addr)
 
 
-    def get_bounds(self):
+    def get_arena_for_chunk(self,addr):
+        chunk = pwndbg.memory.poi(self.malloc_state,addr)
+        _,_,nm = self.chunk_flags(chunk['size'])
+        if nm:
+            r=self.get_arena(arena_addr=self.get_heap(addr)['ar_ptr'])
+        else:
+            r=self.main_arena
+        return r
+            
+    def get_region(self,addr=None):
         """
-        Finds the heap bounds by using mp_ structure's sbrk_base property
+        Finds the memory region used for heap by using mp_ structure's sbrk_base property
         and falls back to using /proc/self/maps (vmmap) which can be wrong
         when .bss is very large
+        For non main-arena heaps use heap_info struct provieded at the beging of the page.
         """
-        lower, upper = None, None
 
-        try:
-            lower = int(self.mp['sbrk_base'])
-        except:
-            lower = None
+        if addr:
+            heap  = self.get_heap(addr)
+            base  = int(heap.address) + self.heap_info.sizeof + self.malloc_state.sizeof
+            page  = pwndbg.vmmap.find(base)
+            ## trim whole page to look exactly like heap
+            page.size = heap['size']
+            page.vaddr = base
+            
+            return page
 
+
+
+        lower = self.mp.get('sbrk_base',None)
```

Comment:
Not sure if you were going to make more changes, but currently we can't assume it is a dict in here.
```gdb
dc@dc $ cat main.c 
#include <stdlib.h>

void foo() { malloc(0xdead); }
void bar() { malloc(1234); }

int main() {
  foo();
  bar();
}
dc@dc $ gcc main.c
dc@dc $ gdb -q ./a.out
Loaded 107 commands. Type pwndbg [filter] for a list.
Reading symbols from ./a.out...(no debugging symbols found)...done.
pwndbg> set context-sections ''
Set which context sections are displayed by default (also controls order) to ''
pwndbg> b bar
Breakpoint 1 at 0x40053b
pwndbg> r
Starting program: /home/dc/a.out 

Breakpoint 1, 0x000000000040053b in bar ()
Breakpoint bar
pwndbg> heap
Traceback (most recent call last):
  File "/home/dc/pwndbg/pwndbg/commands/__init__.py", line 56, in __call__
    return self.function(*args, **kwargs)
  File "/home/dc/pwndbg/pwndbg/commands/__init__.py", line 130, in _OnlyWhenRunning
    return function(*a, **kw)
  File "/home/dc/pwndbg/pwndbg/commands/heap.py", line 61, in heap
    heap_region = main_heap.get_region(addr)
  File "/home/dc/pwndbg/pwndbg/heap/ptmalloc.py", line 192, in get_region
    lower = self.mp.get('sbrk_base',None)
AttributeError: 'gdb.Value' object has no attribute 'get'

pwndbg> 
```